pkg/recipes/configloader/environment_test.go

```diff
@@ -177,6 +177,53 @@ func TestGetConfiguration(t *testing.T) {
 			},
 			errString: ErrUnsupportedComputeKind.Error(),
 		},
+		{
+			name: "recipe config with env resource",
+			envResource: &model.EnvironmentResource{
+				Properties: &model.EnvironmentProperties{
+					Compute: &model.KubernetesCompute{
+						Kind:       to.Ptr(kind),
+						Namespace:  to.Ptr(envNamespace),
+						ResourceID: to.Ptr(envResourceId),
+					},
+					RecipeConfig: &model.RecipeConfigProperties{
+						Terraform: &model.TerraformConfigProperties{
+							Authentication: &model.AuthConfig{
+								Git: &model.GitAuthConfig{
+									Pat: map[string]*model.Secret{
+										"dev.azure.com": &model.Secret{
+											SecretStore: to.Ptr("secretStoreID"),
+										},
+									},
+								},
+							},
+						},
+					},
```

Comment:
We should have a few cases that would cause errors. Like what happens if we remove the PAT?